abi.html

```diff
@@ -6426,57 +6440,134 @@ <h4><a href="#vague-static"> 5.2.2 Static Data </a></h4>
+containing the definition.
+
+<blockquote>
+  <span class="future-abi">Recommendation for new platforms: when
+  selecting the key function, ignore virtual function definitions
+  that are inline outside of the class definition.  Inline function
+  definitions must be defined in every translation unit that ODR-use
+  them, and virtual functions are considered to always be ODR-used.
+  Therefore, in a valid program, every translation unit that sees
+  a class definition must agree about which of its virtual functions
+  are defined inline outside of the class.  Some platforms are known
+  to have already adopted this rule.</span>
+</blockquote>
+
+<blockquote>
+  <span class="future-abi">Recommendation for new platforms:
+  when selecting the key function, also consider pure virtual
+  destructors as candidates.  Virtual destructors must be defined
+  even if they are pure.</span>
+</blockquote>
 
 <p>
 <a name="vague-rtti">
-<h4><a href="#vague-rtti"> 5.2.4 Typeinfo</a></h4>
-
-<p>
-The RTTI std::type_info structure for a complete class type is
-emitted in the same object as its virtual table if dynamic,
-or everywhere referenced if not.
-The RTTI std::type_info structure for an incomplete class type is
-emitted wherever referenced.
-The RTTI std::type_info structures for various basic types as
-specified by the <a href=#rtti>Run-Time Type Information</a> section
-are provided by the runtime library.
-The RTTI name NTBS objects are emitted with each referencing
-std::type_info object.
-
-<p>
-The RTTI std::type_info structures for complete class types and basic
-types are emitted in COMDAT groups identified by their mangled names.
-The RTTI std::type_info structures for incomplete class types
-are emitted with other than the ABI-defined complete type mangled names;
-an implementation may choose to emit them as local static objects,
-or in COMDAT groups with implementation-defined names and COMDAT identifiers.
-The RTTI name NTBS objects are emitted in separate COMDAT groups
-identified by the NTBS mangled names
-as weak symbols.
+<h4><a href="#vague-rtti"> 5.2.4 <code>std::type_info</code></a></h4>
+
+<p>
+The RTTI <code>std::type_info</code> structure for a complete dynamic
+class type whose virtual tables are emitted in a unique object (see
+<a href="#vague-vtable">above</a>) is emitted with non-vague linkage
+in the same object as its virtual tables.
+
+<p>
+The RTTI <code>std::type_info</code> structure for an incomplete class
+type is emitted wherever referenced.  It is implementation-defined
+what symbol name it uses, other than that it must not be the same
+symbol name as any other entity, including the complete class type.
+It also implementation-defined whether they are emitted in a COMDAT
+group or just with internal linkage.
+
+<p>
+The RTTI <code>std::type_info</code> structures for various basic
+types as specified by the <a href=#rtti>Run-Time Type Information</a>
+section are provided by the runtime library.
```

Comment:
Is it worth documenting that we expect these to get emitted in the same object as the vtable for `__cxxabiv1::__fundamental_type_info`?

Reply:
Eh, I feel like that's a runtime-library implementation detail that doesn't need to be in the ABI.